specifications/Dutch Auction.md

```diff
@@ -0,0 +1,172 @@
+---
+title: Dutch Auction Smart Contract
+author: Gustavo Vazquez
+discussion: 
+team: Backend
+category: Development
+status: Draft
+related: https://github.com/alicenet/alicenet/issues/398
+created: 2022-12-07
+---
+
+## Introduction
+
+#### Summary   
+Stakers looking to become a validator should be able to bid for joining price in an dutch style auction.
+The price curve should take into account the need for new validator in both the starting price and the rate of change.
+
+#### Context
+AliceNet validator nodes communicate with each other in a secure way through the use of a distributed key.
+The operation through which this distributed key is generated is called ETHDKG (Ethereum Distribute Key Generation) ceremony.
+
+When a new validator is admitted to join AliceNet, two ETHDKG ceremonies must be executed:
+* One to destroy the current distributed key (all current validators)
+* One to generate a new distributed key (all current validators plus the new one)
+
+Since this operation involves interacting with multiple Ethereum Smart Contract functions it is a costly operation estimated in approx 2.5M gas units for each validator.
+To encourage current validators incur in this expense whenever a new validator wants to join the network, an amount to be paid by the new validator will be required.
+This amount will be determined by an auction among the postulants, the one that wins pays the price and is allowed to join the network, the minimum price for this auction must cover at least the expenses of all current validators.
+
+Regular auction systems need to maintain the auction open for a fixed period of time, adding complexity and creating sometimes unnecessary waits. There is a simpler and faster auction system called Dutch Auction where the bidding initial price decreases automatically with time following a parameterized curve until a bid is placed, at this point the bidder wins and the auction is closed.
+
+#### Goals
+Establish a mechanism to determine the price to become a validator.
+
+#### Non Goals
+The mechanism to become a validator.
+
+#### Assumptions
+Values defined at deployment are defined as a guide and can be modified to adapt the price curve to required results.
+
+## Specification
+
+#### Overview
+The solution consists of a Smart Contract that provides the following functionality:
+##### Start an auction
+This operation allows to start an auction by calculating the initial and final prices and registering the current block number as auction's start block, upon execution, an event with all auction details will be emitted.
```

Comment:
Should include what triggers starting an auction.

Reply:
Added